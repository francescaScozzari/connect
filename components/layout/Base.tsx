import { ThemeProvider } from 'styled-components'
import React from 'react'

import { GlobalStyle } from '@/styles/GlobalStyle'
import { Theme } from '@/models/Utils'
import themes from '@/styles/themes'

type Props = {
  children: React.ReactNode
}

const Layout = ({ children }: Props) => {
  return (
    <ThemeProvider theme={themes[Theme.light]}>
      <GlobalStyle />
      {children}
    </ThemeProvider>
  )
}

export default Layout
