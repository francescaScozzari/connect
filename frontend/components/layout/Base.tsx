import React from 'react'
import { ThemeProvider } from 'styled-components'
import { ToastContainer } from 'react-toastify'

import { GlobalStyle } from '@/styles/GlobalStyle'
import { Theme } from '@/models/Utils'
import { Footer } from '@/components/layout/Footer'
import themes from '@/styles/themes'

type Props = {
  children: React.ReactNode
}

const Layout = ({ children }: Props) => {
  return (
    <ThemeProvider theme={themes[Theme.light]}>
      <ToastContainer />
      <GlobalStyle />
      {children}
      <Footer />
    </ThemeProvider>
  )
}

export default Layout
