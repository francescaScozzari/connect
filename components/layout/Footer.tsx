import React from 'react'
import { styled } from 'styled-components'

const Footer = () => {
  return (
    <OuterWrapper>
      <FooterLink data-cy="uda" href="https://en.unich.it/">
        © 2023 All rights reserved Ud'A - D’Annunzio University of
        Chieti-Pescara
      </FooterLink>
      <InnerWrapper>
        <FooterLinkVariant>Privacy Policy</FooterLinkVariant>
        <FooterLinkVariant
          data-cy="ingenium"
          href="https://ingenium-eu.education/"
        >
          Ingenium University
        </FooterLinkVariant>
        <FooterLinkVariant data-cy="bi4e" href="https://boosting-ingenium.com/">
          BI4E
        </FooterLinkVariant>
      </InnerWrapper>
    </OuterWrapper>
  )
}

const OuterWrapper = styled.footer`
  min-height: 4.5em;
  padding: 2em 4.5em;
  display: flex;
  justify-content: space-between;
  background-color: ${({ theme }) => theme.colors.primary[0]};

  @media (max-width: 1280px) {
    padding: 2em 3.5em;
  }

  @media (max-width: 768px) {
    padding: 2em 2em;
  }
`

const InnerWrapper = styled.footer`
  display: flex;
  gap: 2em;

  @media (max-width: 1280px) {
    gap: 1.5em;
  }

  @media (max-width: 768px) {
    gap: 0.5em;
  }
`

const FooterLink = styled.a`
  color: white;
  font-size: 0.75rem;
  text-transform: capitalize;
`

const FooterLinkVariant = styled(FooterLink)`
  color: #c9d2ec;
  text-decoration: underline;
  font-weight: 500;
`

export { Footer }
