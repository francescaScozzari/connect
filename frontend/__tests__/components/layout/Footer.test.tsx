import { jest, expect } from '@jest/globals'
import React from 'react'

import { Footer } from '@/components/layout/Footer'
import { renderWithWrappers } from '@/__tests__/functions'

const setup = () => renderWithWrappers(<Footer />)

describe('<Footer />', () => {
  test('Footer renders correctly', () => {
    const container = setup()
    expect(container.firstChild).toMatchSnapshot()
  })
})
