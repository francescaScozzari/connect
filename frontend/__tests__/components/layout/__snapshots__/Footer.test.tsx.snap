// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Footer renders correctly 1`] = `
.c0 {
  min-height: 4.5em;
  padding: 2em 4.5em;
  display: flex;
  justify-content: space-between;
  background-color: #2B2D42;
}

.c2 {
  display: flex;
  gap: 2em;
}

.c1 {
  color: white;
  font-size: 0.75rem;
  text-transform: capitalize;
}

.c3 {
  color: #c9d2ec;
  text-decoration: underline;
  font-weight: 500;
}

@media (max-width: 1280px) {
  .c0 {
    padding: 2em 3.5em;
  }
}

@media (max-width: 768px) {
  .c0 {
    padding: 2em 2em;
  }
}

@media (max-width: 1280px) {
  .c2 {
    gap: 1.5em;
  }
}

@media (max-width: 768px) {
  .c2 {
    gap: 0.5em;
  }
}

<footer
  class="c0"
>
  <a
    class="c1"
    data-cy="uda"
    href="https://en.unich.it/"
  >
    © 2023 All rights reserved Ud'A - D’Annunzio University of Chieti-Pescara
  </a>
  <footer
    class="c2"
  >
    <a
      class="c1 c3"
      data-cy="privacy-policy"
      href="https://example.com/connect-privacy-policy"
    >
      Privacy Policy
    </a>
    <a
      class="c1 c3"
      data-cy="ingenium"
      href="https://ingenium-eu.education/"
    >
      Ingenium University
    </a>
    <a
      class="c1 c3"
      data-cy="bi4e"
      href="https://boosting-ingenium.com/"
    >
      BI4E
    </a>
  </footer>
</footer>
`;
